airqo-platform/AirQo-api
# chore: device monitoring data segregation

- What does this PR do?
This PR adds multi-tenancy to the `device-monitoring` microservice.
- What JIRA task is related to this PR?
The link to the JIRA board task is [here](https://airqoteam.atlassian.net/browse/PLAT-73).
- How do I test out this PR?
Git clone this repo `cd AirQo-api/src/device-monitory` . And then run the following `docker` commands
`docker build -t test-device-monitor .` and `docker run -it -p 5555:4001 test-device-monitor`. `OR:` You can create and activate a python virtual env to run the service. Using: `python3 -m venv env`, `source env/bin/activate` and `pip install -r requirements.txt`. Now run `flask run` to test out the points using any REST client of your choice.
- Which end points should are ready for testing?
All endpoints as shown [in this documentation](https://docs.google.com/spreadsheets/d/1-NGTDBtTcIi8Fg2UYSWjcK6geKrZdopz5kxRFsCEMPs/edit#gid=1063367751) You just need to add the tenant parameter to each url e.g. http://localhost:5555/device/monitor/status?tenant=airqo or `?tenant=kcca`
